test/src/auxkernels/OptionalTestAux.C

```diff
@@ -0,0 +1,48 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "OptionalTestAux.h"
+
+registerMooseObject("MooseTestApp", OptionalTestAux);
+
+InputParameters
+OptionalTestAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addRequiredParam<MaterialPropertyName>("prop", "optional non-AD property");
+  params.addRequiredParam<MaterialPropertyName>("adprop", "optional AD property");
+  params.addRequiredParam<bool>("expect", "expect non-AD property to exist");
+  params.addRequiredParam<bool>("adexpect", "expect AD property to exist");
+  return params;
+}
+
+OptionalTestAux::OptionalTestAux(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _prop(getOptionalMaterialProperty<Real>("prop")),
+    _adprop(getOptionalADMaterialProperty<Real>("adprop")),
```

Comment:
For some reason... these calls aren't showing up in coverage?

Reply:
Yeah, weird